zx/compare.go

```diff
@@ -199,7 +199,7 @@ func CompareFloat64(op string, pattern float64) (Predicate, error) {
 	}
 	return func(val zng.Value) bool {
 		zv := val.Bytes
-		switch val.Type.(type) {
+		switch zng.AliasedType(val.Type).(type) {
```

Comment:
You should be using `val.Type.ID()` instead of this (at least this is the convention that @mccanne wanted).

Reply:
I don't have any objection to changing but:
1. We have tons of places in the code that use the `(type)` technique, if we want to make this change, that seems like something orthogonal that should be done through the tree in a separate PR
2. What's the rationale for changing?

Reply:
1. All the alias changes that I made use `Type.ID()` so if we should decide to go this route we should change those as well (in another pr of course).
2. Probably better @mccanne to explain, but I'll take my best shot. I personally find it confusing that primitive types are instances of `TypeOf*` whereas records/sets/arrays don't follow this pattern (for good reasons). A better reason: With aliases functionality, now every time you want to access a value's type you'll have to unwrap it first; I imagine that it would be easy to forget about why this needs to be done. If you forget to unwrap the functionality would appear to generally work except that one time you come across an alias. The convention to always do type switching using `Type.ID()`, seems less error prone and requiring less institutional knowledge from  contributors/reviewers.